application/client.core/src/app/environment/components/sidebar/shell/input/template.html

```diff
@@ -4,40 +4,37 @@
-                    type="text"
-                    placeholder="Type a command"
-                    aria-label="Filter"
-                    (focus)="_ng_onFocusRequestInput()"
-                    (blur)="_ng_onBlurRequestInput()"
-                    (keyup)="_ng_onKeyUpRequestInput($event)"
-                    (keydown)="_ng_onKeyDownRequestInput($event)"
-                    (click)="_ng_onClickRequestInput()"
-                    matInput
-                    [formControl]="_ng_inputCtrl"
-                    matAutocomplete
-                    [matAutocomplete]="auto">
-            <mat-autocomplete #auto="matAutocomplete" class="regular"
-                (optionSelected)="_ng_onRecentSelected($event)"
-                (opened)="_ng_onAutocompletePanelOpen()"
-                [classList]="'recent-requests'"
-                [disableRipple]="true">
-                <mat-option *ngFor="let recent of _ng_recent | async" [value]="recent" [attr.title]="recent" class="recent-requests small t-console t-small">
+    <mat-form-field class="recent-requests outline regular no-margin" appearance="outline">
+        <input  #requestinput
+                type="text"
+                placeholder="Type a command"
+                aria-label="Filter"
+                (focus)="_ng_onFocusRequestInput()"
+                (blur)="_ng_onBlurRequestInput()"
+                (keyup)="_ng_onKeyUpRequestInput($event)"
+                (keydown)="_ng_onKeyDownRequestInput($event)"
+                (click)="_ng_onClickRequestInput()"
+                matInput
+                [formControl]="_ng_inputCtrl"
+                matAutocomplete
+                [matAutocomplete]="auto">
+        <mat-autocomplete #auto="matAutocomplete" class="regular"
+            (optionSelected)="_ng_onRecentSelected($event)"
+            (opened)="_ng_onAutocompletePanelOpen()"
+            [classList]="'recent-requests options'"
+            [disableRipple]="true">
+            <mat-option *ngFor="let recent of _ng_recent | async"
+                        [value]="recent"
+                        [attr.title]="recent"
+                        class="recent-requests small t-console t-small"
+                        (mouseenter)="_ng_onMouseOver(recent.description)"
+                        (mouseleave)="_ng_onMouseOver('')">
+                <div class="options">
                     <span class="filename" [innerHTML]="_ng_getSafeHTML(recent.tdescription)"></span>
-                </mat-option>
-            </mat-autocomplete>
-        </mat-form-field>
-    </div>
-    <div class="buttons">
-        <button mat-stroked-button class="button" color="primary" (click)="_ng_onClearRecent()">Clear</button>
-        <button mat-stroked-button class="button" color="primary" (click)="_ng_onEnvironment()">Environment</button>
-    </div>
+                    <mat-icon *ngIf="_ng_hoverRecent === recent.description" (click)="_ng_onRemove(recent.description)">close</mat-icon>
```

Comment:
Add/remove node in runtime quite expensive. We can resolve this using just css. I suggest get rid of `*ngIf="_ng_hoverRecent === recent.description"` but on css level do something like
```
.parent {
   & .icon {
       display: none;
   }
   &:hover {
        & .icon {
            display: inline-block;
        }
   }
}

```